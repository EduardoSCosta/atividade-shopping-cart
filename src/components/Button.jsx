import styled from 'styled-components';

const getButtonVariantStyles = (variant, theme) => {
  const variantStyles = {
    blue: {
      backgroundColor: theme.colors.blue300,
      color: theme.colors.white
    },
    red: {
      backgroundColor: theme.colors.red300,
      color: theme.colors.black
    },
    green: {
      backgroundColor: theme.colors.green300,
      color: theme.colors.black
    },
    white: {
      backgroundColor: theme.colors.white,
      color: theme.colors.blue300
    }
  }

  return variantStyles[variant] || variantStyles.blue
}

export const Button = styled.button({
  padding: ({ theme }) => theme.space[3],
  textTransform: 'uppercase',
  border: 'none',
  borderRadius: ({ theme }) => theme.radii[1],
  backgroundColor: ({ variant, theme }) => getButtonVariantStyles(variant, theme).backgroundColor,
  color:({ variant, theme }) => getButtonVariantStyles(variant, theme).color,
  transition: 'filter 0.2s',

  '&:hover:enabled': {
    filter: 'brightness(0.8)'
  },

  '&:disabled': {
    backgroundColor: ({ theme }) => theme.colors.gray200,
  },

});
