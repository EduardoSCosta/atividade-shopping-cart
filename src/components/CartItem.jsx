import '../styles/components/_cartItem.css';

const CartItem = ({ cartItem, removeFromCart }) => {
  return (
    <div className='cart-item' data-testid='cart-item'>
      <p className='cart-item__name'>{cartItem.name}</p>
      <p className='cart-item__description'>{cartItem.description}</p>
      <button className='button cart-item__remove-from-cart-button' type='button' onClick={removeFromCart}>
        Remove from cart
      </button>
    </div>
  );
}

export default CartItem;
