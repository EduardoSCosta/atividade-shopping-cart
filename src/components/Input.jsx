import styled from 'styled-components';

const getInputWrapperVariantStyles = (variant, theme) => {
  const variantStyles = {
    filled: {
      border: 'none',
      backgroundColor: theme.colors.gray200,
      transition: 'filter 0.2s',
      '&:hover': {
        filter: 'brightness(0.95)'
      },
    },
    outlined: {
      border: `1px solid ${theme.colors.gray200}`,
      '&:hover': {
        borderColor: theme.colors.black
      },
    },
    standard: {
      backgroundColor: 'white',
      borderBottom: `1px solid ${theme.colors.black}`,
      '&:hover': {
        borderBottom: '2px solid'
      },
    },
  };
  return variantStyles[variant] || variantStyles.standard;
}

const InputWrapper = styled.div({
  display: 'flex',
  flexDirection: 'column',
  width: ({ fullWidth }) => fullWidth ? '100%' : 'auto',
  height: ({ theme }) => theme.sizes[0],
  border: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme).border,
  borderBottom: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme).borderBottom,
  backgroundColor: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme).backgroundColor,
  transition: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme).transition,

  '&:hover': {
    filter: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme)['&:hover'].filter,
    borderColor: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme)['&:hover'].borderColor,
    borderBottom: ({ variant, theme }) => getInputWrapperVariantStyles(variant, theme)['&:hover'].borderBottom
  }
}
);

const StyledLabel = styled.label({
  color: ({ theme }) => theme.colors.black,
  paddingLeft: ({ theme }) => theme.space[3]
});

const StyledInput = styled.input({
  border: 'none',
  height: '100%',
  backgroundColor: 'transparent'
});

const Input = ({
  required,
  id,
  name,
  label,
  fullWidth = false,
  autoComplete,
  variant,
  value,
  onChange }) => {

  return (
    <InputWrapper variant={variant} fullWidth={fullWidth}>
      <StyledLabel htmlFor={name}>{label}</StyledLabel>
      <StyledInput
        required={required}
        id={id}
        name={name}
        autoComplete={autoComplete}
        value={value}
        onChange={onChange}
      />
    </InputWrapper>
  );
}

export default Input;
