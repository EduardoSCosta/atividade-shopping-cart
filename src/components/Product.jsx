import '../styles/components/_product.css';

const Product = ({product, addToCart}) => {
  return (
    <div className='product' data-testid='product'>
      <p className='product__name'>{product.name}</p>
      <p className='product__description'>{product.description}</p>
      <button className='button product__add-to-cart-button' type='button' onClick={addToCart}>
        Add to cart
      </button>
    </div>
  );
}

export default Product;
