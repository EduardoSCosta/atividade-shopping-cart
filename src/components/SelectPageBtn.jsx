import '../styles/components/_selectPageBtn.css'

const SelectPageButton = ({pageName, setCurrentPage}) => {
  return (
    <button className='button button--select-page' type='button' onClick={() => setCurrentPage(pageName)}>
      {pageName}
    </button>
  );
}

export default SelectPageButton;
