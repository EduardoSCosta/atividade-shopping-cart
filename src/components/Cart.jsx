import '../styles/components/_cart.css'
import CartItem from './CartItem';
import { Button, Text } from '../components';
import { css } from '@styled-system/css';

const Cart = ({ cartItems, removeFromCart, goToCheckout }) => {
  return (
    <div className='page'>
      <Text variant='h2'
        css={css({
          textAlign: 'center',
          marginY: [4]
        })}
      >
        Cart
      </Text>
      <div className='cart-items-list'>
        {cartItems.map((cartItem) => {
          return (
            <CartItem key={cartItem.id} cartItem={cartItem} removeFromCart={() => removeFromCart(cartItem)} />
          );
        })}
      </div>
      <Button
        variant='blue'
        css={css({
          display: 'block',
          marginX: 'auto',
          marginY: [3]
        })}
        type='button'
        onClick={goToCheckout}
      >
        Checkout
      </Button>
    </div>
  );
}

export default Cart;
