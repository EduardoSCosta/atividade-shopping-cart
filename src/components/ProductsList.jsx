import Product from './Product';

const ProductsList = ({ products, addToCart }) => {
  return (
    <div className='products-list'>
      <h2>Products List</h2>
      {products.map((product) => {
        return (
          <Product key={product.id} product={product} addToCart={() => addToCart(product)} />
        );
      })}
    </div>
  );
}

export default ProductsList;
