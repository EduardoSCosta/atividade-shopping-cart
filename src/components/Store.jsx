import { useState } from 'react';
import Cart from './Cart';
import ProductsList from './ProductsList';
import SelectPageButton from './SelectPageBtn';
import '../styles/components/_store.css'

const PRODUCTS_LIST = [
  { id: 1, name: 'product one', description: 'This is product one.' },
  { id: 2, name: 'product two', description: 'This is product two.' },
  { id: 3, name: 'product three', description: 'This is product three.' },
  { id: 4, name: 'product four', description: 'This is product four.' }
]
const PRODUCTS_PAGE = 'products';
const CART_PAGE = 'cart';

const Store = () => {
  const [products, setProducts] = useState(PRODUCTS_LIST);
  const [cartItems, setCartItems] = useState([]);
  const [currentPage, setCurrentPage] = useState(PRODUCTS_PAGE);

  const isCurrentPage = currentPage === PRODUCTS_PAGE;

  const addToCart = (product) => {
    const isAlreadyInCart = cartItems.some(cartItem => cartItem.id === product.id);

    if (!isAlreadyInCart) {
      setCartItems([...cartItems, product]);
    }
  }

  const removeFromCart = (product) => {
    setCartItems(cartItems.filter(cartItem => cartItem.id !== product.id));
  }

  return (
    <>
      <header className='header'>
        <SelectPageButton pageName={PRODUCTS_PAGE} setCurrentPage={setCurrentPage} />
        <SelectPageButton pageName={CART_PAGE} setCurrentPage={setCurrentPage} />
      </header>
      {(isCurrentPage) ?
        <ProductsList products={products} addToCart={addToCart} />
        :
        <Cart cartItems={cartItems} removeFromCart={removeFromCart} />
      }
    </>
  );
}

export default Store;
