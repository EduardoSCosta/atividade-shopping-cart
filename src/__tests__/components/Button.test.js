import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { Button } from '../../components';
import themes from '../../theme';

const theme = themes.light;

describe('<Button />', () => {
  it('renders the Button component correctly', () => {
    const buttonComponent = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button variant='green'>TestButton</Button>
        </ThemeProvider>)
      .toJSON();
    expect(buttonComponent).toMatchSnapshot();
  })
})
