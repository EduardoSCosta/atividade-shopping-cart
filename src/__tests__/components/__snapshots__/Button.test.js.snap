// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders the Button component correctly 1`] = `
<button
  className="Button-sc-1l27lxk-0 kYrFaz"
>
  TestButton
</button>
`;
